src/transformers/modeling_bart.py

```diff
@@ -0,0 +1,1062 @@
+        **loss**: (`optional`, returned when ``lm_labels`` is provided) ``torch.FloatTensor`` of shape ``(1,)``:
+            Masked language modeling loss.
+        **prediction_scores**: ``torch.FloatTensor`` of shape ``(batch_size, sequence_length, config.vocab_size)``
+            Prediction scores of the language modeling head (scores for each vocabulary token before SoftMax).
+        **hidden_states**: (`optional`, returned when ``config.output_hidden_states=True``)
+            list of ``torch.FloatTensor`` (one for the output of each layer + the output of the embeddings)
+            of shape ``(batch_size, sequence_length, hidden_size)``:
+            Hidden-states of the model at the output of each layer plus the initial embedding outputs.
+        **attentions**: (`optional`, returned when ``config.output_attentions=True``)
+            list of ``torch.FloatTensor`` (one for each layer) of shape ``(batch_size, num_heads, sequence_length, sequence_length)``:
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention heads.
+
+    Examples::
+
+        tokenizer = BartTokenizer.from_pretrained('bart-large')
+        model = BartForMaskedLM.from_pretrained('bart-large')
+        input_ids = torch.tensor(tokenizer.encode("Hello, my dog is cute")).unsqueeze(0)  # Batch size 1
+        outputs = model(input_ids=input_ids, lm_labels=input_ids)
+        loss, prediction_scores = outputs[:2]
+
+    """
+    base_model_prefix = "model"
+
+    def __init__(self, config: BartConfig):
+        super().__init__(config)
+        self.model = BartModel(config)
+        self.lm_head = _make_linear_from_emb(self.model.shared)
+
+    def forward(
+        self,
+        input_ids,
+        attention_mask=None,
+        encoder_hidden_states=None,
+        decoder_input_ids=None,
+        decoder_attention_mask=None,
+        cached_states=None,
+        lm_labels=None,
+        **unused
+    ):
+        outputs = self.model.forward(
+            input_ids,
+            attention_mask=attention_mask,
+            decoder_input_ids=decoder_input_ids,
+            encoder_hidden_states=encoder_hidden_states,
+            decoder_attention_mask=decoder_attention_mask,
+            cached_states=cached_states,
+        )
+        lm_logits = self.lm_head.forward(outputs[0])
+        outputs = (lm_logits,) + outputs[1:]  # Add hidden states and attention if they are here
+        if lm_labels is not None:
+            loss_fct = nn.CrossEntropyLoss()
+            masked_lm_loss = loss_fct(lm_logits.view(-1, self.config.vocab_size), lm_labels.view(-1))
+            outputs = (masked_lm_loss,) + outputs
+
+        return outputs
+
+    @staticmethod
+    def prepare_inputs_for_generation(input_ids, past, **kwargs):
+        return {"input_ids": input_ids, "cached_states": past, "decoder_input_ids": input_ids}
```

Comment:
Maybe would be a good idea to rename `cached_states` to `decoder_cached_states` as the `cached_states` only correspond to the decoder as far as I can see.